ampersand-chart.js

```diff
@@ -912,11 +962,27 @@
       if (this.model.drawXAxisLabels) {
         container.append('text')
           .attr('class', 'ampersand-graph-label')
+          .style('display', 'none')
+          .style('opacity', 0)
           .attr('y', height - 46)
           .attr('dy', '1.25em');
 
-        containers.select('text.ampersand-graph-label')
-          .attr('x', areaWidth / 2);
+        _.defer(function(containers, areaWidth, values, label) {
+          containers.select('text.ampersand-graph-label')
+            .style('opacity', 0)
+            .style('display', function(d, i) {
+              var width = Math.ceil(_.max(data, function(item) { return item[label].length; })[label].length * 6.25);
+              var areaGroupWidth = (areaWidth * values.length);
+              if (width < areaGroupWidth || this.model.drawAllXAxisLabels) {
+                return undefined;
+              }
+
+              return i % Math.ceil(width / areaGroupWidth) === 0 ? undefined : 'none';
```

Comment:
These code blocks all have a lot in common.  It definitely seems like something should be able to be pulled out to reduce that duplication.